lib/relay.js

```diff
@@ -197,37 +250,72 @@ function responseHandler(filterRules, config) {
 
       logger.debug(logContext, 'sending websocket request over HTTP connection');
 
-      request({
+      var req = {
         url: result.url,
         headers: headers,
         method,
         body,
         agentOptions: {
           ca: config.caCert, // Optional CA cert
         },
-      }, (error, response, body) => {
+      };
+
+      // check if this is a streaming request for binary data
+      if (streamingID) {
+        logger.debug(logContext, 'serving stream request');
+
+        req.encoding = null; // no encoding for binary data
+
+        request(req).on('response', response => {
+          const status = (response && response.statusCode) || 500;
+          logResponse(logContext, status, response.headers);
+          io.send('chunk', streamingID, '', false, {
+            status,
+            headers : response.headers
+          });
+        }).on('data', chunk => {
+          io.send('chunk', streamingID, chunk, false);
+        }).on('end', () => {
+          io.send('chunk', streamingID, '', true);
+        }).on('error', error => {
+          logError(logContext, error);
+          io.send('chunk', streamingID, error.message, true, {
+            status : 500
+          });
+        });
+        return;
+      }
+
+      request(req, (error, response, body) => {
```

Comment:
I've tried to merge the two request statements into one but this got quite messy and didn't reduce the code much. I feel that the two mechanisms are distinctive enough to warrant two branches.